pkg/builder/TwoSampleTTestBuilder.go

```diff
@@ -0,0 +1,332 @@
+// and a negative difference in the experimental population would be bad.
+// This information is determined outside of this builder (the builder doesn't know
+// what parameter combination is being tested) so the builder must be told
+// what the "goodnes polarity" is +1 or -1.
+func (scc *ScorecardCell) SetGoodnessPolarity(polarity GoodnessPolarity) error {
+	errs := validate.Var(polarity, "required,oneof=-1 1")
+	if errs != nil {
+		log.Print(errs)
+		return errors.New(fmt.Sprint("TwoSampleTTestBuilder SetGoodnessPolarity", errs))
+	} else {
+		scc.goodnessPolarity = polarity
+	}
+	return nil // no errors
+}
+
+// set the major p-value threshold
+func (scc *ScorecardCell) SetMajorThreshold(threshold Threshold) error {
+	if errs := validate.Var(threshold, "required,gt=0,lt=.5"); errs != nil {
+		log.Print(errs)
+		return errors.New(fmt.Sprint("TwoSampleTTestBuilder SetMajorThreshold", errs))
+	} else {
+		scc.majorThreshold = threshold
+	}
+	return nil // no errors
+}
+
+// set the major p-value threshold
+func (scc *ScorecardCell) SetMinorThreshold(threshold Threshold) error {
+	if errs := validate.Var(threshold, "required,gt=0,lt=.5"); errs != nil {
+		log.Print(errs)
+		return errors.New(fmt.Sprint("TwoSampleTTestBuilder SetMinorThreshold", errs))
+	} else {
+		scc.minorThreshold = threshold
+	}
+	return nil // no errors
+}
+
+// get the return value based on the major and minor thresholds compared to the p-value
+func deriveValue(scc *ScorecardCell, difference float64, pval float64) (int, error) {
+	if errs := validate.Var(difference, "required"); errs != nil {
+		log.Print(errs)
+		return 0, errors.New(fmt.Sprint("TwoSampleTTestBuilder deriveValue", errs))
+	} else {
+		if errs := validate.Var(pval, "required"); errs != nil {
+			fmt.Println(errs)
+			return -9999, errors.New(fmt.Sprint("TwoSampleTTestBuilder deriveValue", errs))
+		} else {
+			if pval <= float64(scc.majorThreshold) {
+				return 2 * int(scc.goodnessPolarity), nil
+			}
+			if pval <= float64(scc.minorThreshold) {
+				return 1 * int(scc.goodnessPolarity), nil
+			}
+			return 0, nil
+		}
+	}
+}
+
+// using the experimental Query Result and the control QueryResult and the statistic
+// perform statistic calculation for each, perform matching and store the resultant  dataSet
+func deriveCTCInputData(scc *ScorecardCell, QR *QueryResult, statisticType string) (dataSet DataSet, err error) {
+	// derive CTC statistical values for ctl and exp
+	var stat float32
+	var ctlData []PreCalcRecord
+	var expData []PreCalcRecord
+	ctlQR := *(QR.CtlData)
+	expQR := *(QR.ExpData)
+	var record CTCRecord
+
+	for i := 0; i < len(ctlQR); i++ {
+		record = (ctlQR)[i].(CTCRecord)
+		stat, err = CalculateStatCTC(record.Hit, record.Fa, record.Miss, record.Cn, statisticType)
+		if err == nil {
+			//include this one
+			ctlData = append(ctlData, PreCalcRecord{Value: float64(stat), Time: record.Time})
+		} else { /*don't include it*/
+		}
```

Comment:
I believe empty branches are frowned on by the Go linter. Instead, the code could do something like:

```suggestion
		if err != nil {
			// skip this value
			continue
		}
		ctlData = append(ctlData, PreCalcRecord{Value: float64(stat), Time: record.Time})
```

This applies to other areas like lines 132-136, etc...